webui/react/src/pages/Cluster/ClusterOverview.tsx

```diff
@@ -82,6 +82,28 @@ const ClusterOverview: React.FC = () => {
     return getSlotContainerStates(agents || [], ResourceType.CPU);
   }, [ agents ]);
 
+  const [ cudaTotalSlots, rocmTotalSlots, cpuTotalSlots ] = useMemo(() => {
+    return resourcePools.reduce((acc, pool) => {
+      let index;
+      switch (pool.slotType) {
+        case ResourceType.CUDA:
+          index = 0;
+          break;
+        case ResourceType.ROCM:
+          index = 1;
+          break;
+        case ResourceType.CPU:
+          index = 2;
+          break;
+        default:
+          index = undefined;
+      }
```

Comment:
This works and you don't have to change this. An alternative is to create a Record map:
```
const resourceTypeIndexMap = {
  [ResourceType.CUDA]: 0,
  [ResourceType.ROCM]: 1,
  [ResourceType.CPU]: 2,
};
const index = resourceTypeIndexMap[pool.slotType];
```

Reply:
I'll keep that in mind!